### PR TITLE
Sort by date

### PR DESCRIPTION
**Pull request by andyw8**
I'm not super happy with the approach for combining the different
params, e.g. page number, filter and sort order, but I think it's good
enough for now.

### CODE REVIEW

File: app/views/books/index.html.erb [hunk 1]
```diff
@@ -1,11 +1,24 @@
...
-  <ul class="books">
-    <% books.each do |book| %>
-      <li><%= link_to book.title, book %></li>
-    <% end %>
-  </ul>
+
+  <table class="books">
+    <thead>
+      <tr>
+        <th><%= link_to("Title", sort_path("title")) %></th>
```
Comment (taylorrf) [inline app/views/books/index.html.erb]: What you think about use "query strings" here on the `link_to` methods? Using it you can remove the `sort_path` helper.
`<%= link_to("Title", sort: "title") %>`
`<%= link_to("Added", sort: "added") %>`

Comment (andyw8) [inline app/views/books/index.html.erb]: The helper is needed to retain the current page number and filter when changing the sort order.

File: app/models/book.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 class Book < ActiveRecord::Base
+  SORT_CRITERIA_ADDED = "added"
```
Comment (andyw8) [inline app/models/book.rb]: I introduced this to make things less fragile. A typo in a string can be easily missed, but a typo in a constant name will make Ruby fail loudly.

**Comment by andyw8** [comment]
Once #11 is merged I will rebase this on master so the `booklist` CSS changes disappear.

File: app/views/books/index.html.erb (hunk 1 above)
Comment (taylorrf) [inline app/views/books/index.html.erb]: Ha, ok. Make sense to keep the page number state.

File: app/models/book.rb (hunk 2 above)
Comment (taylorrf) [inline app/models/book.rb]: ok

**Comment by taylorrf** [comment]
LGTM

**Comment by taylorrf** [comment]
#16 and #22 is okay.

**Pull request merged**